File: C/beginner/1074.c
// https://judge.beecrowd.com/pt/problems/view/1074
// Par ou ímpar

// Libraries
#include <stdio.h>

// Main Function
int main(){
    // Starting code and variables;
    int user_interations, usr_input;

    // Scanning user input
    scanf("%d", &user_interations);

    // Iterating
    for (int i = 1; i <= user_interations; i++){
        scanf("%d", &usr_input);
        if (usr_input > 0){
            if (usr_input % 2 == 0){
                printf("EVEN POSITIVE\n");
            } else {
                printf("ODD POSITIVE\n");
            }
        } else if (usr_input < 0){
            if (usr_input % 2 == 0){
                printf("EVEN NEGATIVE\n");
            } else {
                printf("ODD NEGATIVE\n");
            }
        } else {
            printf("NULL\n");
        }
    }

    // Ending Code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1074.hs
-- https://judge.beecrowd.com/pt/problems/view/1074
-- Par ou Ímpar

-- Importing ReplicateM
import Control.Monad (replicateM_)

-- Main
main = do
    user_interations <- readLn

-- Iterating
    replicateM_ user_interations $ do
        usr_input <- readLn
        putStrLn $ if usr_input > 0
            then if even usr_input
                then "EVEN POSITIVE"
                else "ODD POSITIVE"
            else if usr_input < 0
                then if even usr_input
                    then "EVEN NEGATIVE"
                    else "ODD NEGATIVE"
                else "NULL"

-- By Fernando Serra
-- https://github.com/fernandosserra